### ISSUE TITLE
Exporter authentication

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-specification#1344]
**What are you trying to achieve?**

Specification for configuring authentication of exporters.

**Additional context.**

Currently, the spec only defines how to configure endpoints and TLS certificates for exporters. This generally works well with localhost collectors, but it is also common to need authentication for exporters, for example basic auth or JWT. Authentication is a general concern and is generally not coupled with a specific exporter implementation (though not always), and we should be able to create a general document of what customization exporters must support for normal authentication scenarios.

We would probably have some standard authentication mechanisms that could have first class APIs / environment variables, for example:
- Basic auth
- TLS mutual auth (only implemented for OTLP right now but should be supported by any exporter)
- Fixed token
- Service account signed JWT
Anything else?

And we should mention that exporters should support custom authentication mechanisms programatically if possible. For example
- AWS Sig4

**Comment by anuraaga** [comment]
For reference, this came up from a user asking for basic auth with jaeger exporter - https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2050

**Comment by pureklkl** [comment]
Curious about the progress there, and any plan for Oauth2.0/OIDC?

**Comment by anuraaga** [comment]
I think the collector has implemented authentication extensions that can be used as a basis for a design for SDKs. @jpkrohling what do you think?

**Comment by jpkrohling** [comment]
We do have auth on the collector side, but the design is appropriate for programs, requiring the auth extensions to exist at compile time. For libraries, I would probably have a different design. But the interfaces themselves are quite lean and can serve as inspiration, if they cannot be reused directly.

**Comment by umarkhan17** [comment]
I want to implement Authentication on Java agent side. i already know how to implement Authentication at collector side. Pls advice on how to implement Extensions for Authentication.

**Comment by yellenseram** [comment]
Wondering if there's any progress on this?

**Comment by DGraySh** [comment]
Hi! Is there any update about it after two years?

**Comment by codeboten** [comment]
A discussion in the configuration working group came up about specifying authenticators. Pinging @zeitlinger @MrAlias

**Comment by zeitlinger** [comment]
Yes, https://github.com/open-telemetry/opentelemetry-configuration/issues/257

**Comment by dashpole** [comment]
I'm would be glad to sponsor this!
